components/apimgt/org.wso2.carbon.apimgt.ballerina.gateway/src/main/ballerina/gateway/throttle_filter.bal

```diff
@@ -171,6 +171,68 @@ function generateThrottleEvent(http:Request req, http:FilterContext context, Aut
     requestStreamDto.appTenant = keyValidationDto.subscriberTenantDomain;
     requestStreamDto.apiTenant = getTenantDomain(context);
     requestStreamDto.apiName = getApiName(context);
-    requestStreamDto.properties = "{}";
+
+    json properties = {};
+    string|() remoteAddr = getRemoteAddr(req);
+    if (remoteAddr != null) {
+        string ip = remoteAddr ?: "";
+        properties.ip = ipToLong(ip);
+    }
+    if (getGatewayConfInstance().getThrottleConf().enabledHeaderConditions){
+        string[] headerNames = req.getHeaderNames();
+        foreach headerName in headerNames {
+            string headerValue = untaint req.getHeader(headerName);
+            properties[headerName] = headerValue;
+        }
+    }
+    if (getGatewayConfInstance().getThrottleConf().enabledQueryParamConditions){
+        foreach k, v in req.getQueryParams() {
+            properties[k] = v;
+        }
+    }
+    requestStreamDto.properties = properties.toString();
     return requestStreamDto;
 }
+
+function getRemoteAddr(http:Request request) returns (string?) {
+    string X_FORWARDED_FOR = "X-Forwarded-For";
+    string|() remoteAddr = null;
```

Comment:
Check if using null is correct

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/5384